Feature/share proposal detail(wr9 113)

HeoJeongHyeon opened the pull request:
# Pull Request Template

## 📝 PR 설명

- 공유 상세 조회 API

## 🔍 주요 변경사항

- testCode 작성
- /api/share-proposals 통일
- 공유 제안 ID를 통해 특정 공유 제안에 대한 편지 내역 조회
- 현재 사용자와 공유 요청 수신자가 일치하는지 확인


## 📸 스크린샷 (선택사항)

- UI 변경사항이 있다면 스크린샷을 첨부해주세요.

## 🧪 테스트

- [x] 단위 테스트 추가/수정
- [x] 테스트 커버리지 확인
- [x] 소나클라우드 품질 게이트 통과

## ✅ 체크리스트

- [x] 코드 컨벤션을 준수했나요?
- [x] 불필요한 코드나 주석이 없나요?
- [x] 브랜치 네이밍 컨벤션을 따랐나요?
- [x] 관련 문서를 업데이트했나요?

## 🤝 관련 이슈

- closes #이슈번호

HeoJeongHyeon committed:
test 공유 요청 상세 조회

- 공유 요청시 수신자와 현재 사용자 체크
- 해당 공유 요청 ID 체크

HeoJeongHyeon committed:
feat 공유 요청 상세 조회

- 공유 요청시 수신자와 현재 사용자 체크
- 해당 공유 요청 ID 체크
- URI share-proposals로 변경

HeoJeongHyeon committed:
refactor 컨트롤러 응답 메시지 변경

재수정

leeys9423 reviewed (commented):
수고하셨습니다

leeys9423 commented on src/main/java/io/crops/warmletter/domain/share/repository/ShareProposalRepositoryImpl.java:
Q클래스 사용 방식 통일하면 좋을 것 같습니다!
현재 코드는 정적 import와 명시적 인스턴스 생성이 혼합되어 있어 일관성이 떨어집니다!
복잡한 조인이 많은 이 코드에서는 명시적 인스턴스 생성 방식으로 통일하는 것이 좋아보입니다~

필요한 경우에만 메서드 내에서 추가 Q클래스 인스턴스를 생성하고, 가능하면 클래스 필드 사용하기

HeoJeongHyeon committed:
refactor Repo 수정

- CustomSharePostRepositoryImpl
QSharePost.sharePost 명시적 생성 수정
- ShareProposalRepository
SHARE_PROPOSAL같은 공통 Q클래스 상수 추가

HeoJeongHyeon committed:
refactor join PR리뷰 관해 업데이트

- leftjoin -> join으로 수정
join으로 수정하여도 shareProposalLetter는 반드시 1개 이상의 공유 편지가 존재한다.
null 값이 들어갈 수 없기에 이부분은 join으로 수정.
